### ISSUE TITLE
Removed as 'Not An Issue'.

### ISSUE THREAD

**Issue by Robby262** [LibreTime/libretime#199]
Danny has a foot in Airtime but is really a Liquid Soap guy.
If we looked at what the output formats that are needed for Icecast and ingested those formats as 'icecast ready', LS caching becomes obsolete, real-time trans-coding becomes obsolete. I guess you could dump liquid soap and do it with just a bit of Sox (a ffmpeg wrapper) or raw ffmpeg or the Ubuntu (Debian) preferred Gstreamer.
If you do not, someone will. It is the logical route forward.

**Comment by Robbt** [comment]
I'm sorry but this comment is not an issue. It doesn't address a problem and it isn't very clear. You make a few assertions that you think LiquidSoap is unnecessary but it really doesn't conform to what we are shooting for in regards to a properly formatted Github issue. 

Just to communicate a little bit. Liquidsoap does more than transcoding, it provides a means for show authentication as well as stream relaying, ie listening to a webstream and rebroadcasting it. If you want to try to develop a LibreTime fork that works without LiquidSoap go for it but I just had to chime in because this we should really try to maintain a little bit of formatting when we are creating issues. See the guidelines or just look at Bill Burtons posts for pointers. The most important thing is that issues should reference a specific problem and provide analysis and/or solutions for this problem. This is just a musing and comment as to your thought process without a grounding in anything specific to the project.

**Comment by bburton** [comment]
@Robby262 Do you have any links that address your points so there's something more concrete to work from? What does "Icecast ready" mean and where is it documented? What are the objections to using Liquidsoap?

The thing is Liquidsoap is an important part of PyPo (Python Playout engine) used to perform all the following tasks:

* Receives a list all the scheduled tracks for each show from the Python playout service and plays them.
* Notifies the Python playout service when a new track starts to play.
* Accepts connections from Master and Show Sources and seamlessly switches between them and the scheduled track input when instructed by the web UI.
* Directs output to one to four streams with different codecs.
* Records audio from sound card.

As far as I know, the software you mentioned can only take one input and send to one output and doesn't support programmatic input stream switching, etc.

Replacing Liquidsoap would likely require a major redesign of PyPo and is not a simple thing.

**Comment by Robby262** [comment]
JohnnyC suggested a forum for points like this that are not really issues, but rather discussion topics. Haremare said that would be a bad thing and that EVERYTHING should be put in HERE as issues, so I have and that seems wrong too. Johnny made an alternative forum, war broke out and he was banned (and still is). Anyway, I can't speak for him, he is not here. I am in the same dev team as him and do pass on some of his 'wisdom'. Currently he is rather sick, unavailable and I am holding the fort. I know he was doing some experiments that were non Pypo, non-LS and not Airtime based. Play-out server loads were tiny.

**Comment by Robby262** [comment]
I shall refrain from further comments.

**Comment by bburton** [comment]
@Robby262 What I suggest is changing the title to something like, "Ideas on replacing liquidsoap" or similar. Then paste in some of the main points you found with links. This would provide some value for future reference.

**Comment by Robby262** [comment]
I don't think so Bill. It will not be appreciated. I saw what they did to my boss. Best just to let it lie.

**Comment by Robbt** [comment]
Look, first off as far as I am aware, and I've read all of the threads. Johnny C was not banned, he self-exiled himself due to a disagreement with @hairmare. The only decision that was made was one of his posts was moderated. Also we were in the process of working on building a forum when he decided that it was taking to long, created his own and then deleted the posting linking to it after his disagreement with hairmare. Your welcome to speculate and use the github issues to do so it would just be helpful if it followed a more appropriate format so that when people are browsing the issues it has more meaning than simply referencing Daniel James. I'm sorry to hear Johnny C is having health problems and I hope he recovers soon. 

Anyways, I didn't mean to come off so harshly that you would feel unwelcome in sharing your ideas etc. Since we don't have a forum you are welcome to post things just try to use the subject and perhaps put Idea: (summary of the idea). Just wanted to clarify.

Later,
   Robb

**Comment by Robby262** [comment]
Untrue. He is still banned and cannot log in.

**Comment by Robby262** [comment]
I will pass on your good wishes. Both JC and his wife are being treated for cancer. I am sure we all wish them well.

**Comment by Robby262** [comment]
Here is the link that got 'moderated'
[http://web-radio-support.websitesinternational.net/](url)

**Comment by Robby262** [comment]
OK I think that is team 'Websites International' and our family of stations done here, at least for now. JC may reverse this.
If and when we see some maturity, structure, quality control, accountability.. who knows.

**Comment by Robby262** [comment]
Unwatching this Git Bye

**Comment by Robby262** [comment]
Unwatching this Git Bye. As has been proven above, the moderators cannot be trusted.

**Comment by Robby262** [comment]
You do realise my long post was culled?

**Comment by Robby262** [comment]
The one that showed the untruths of 2 moderators?

**Comment by Robby262** [comment]
This one?
Since we don't have a forum..
WE MADE A FORUM FOR LIBRETIME - IT WORKS! (been up for a month) - moderated out of existence in this Git!

We did provide a forum as a contribution to the project and BEFORE Haremare suggested making one, which was a reaction to our suggestion. Check the Datestamps. Ours is operational. I wrote it. Where is the Libretime one? I will happily redirect all traffic to it? How would cutting the only link to this site, that he paid for, benefit him? Here is the link that got 'moderated' and lit the fuse..

JohnnyC1951 commented on 11 Apr
[Deleted]

This HAD said: http://web-radio-support.websitesinternational.net/

JC was not best pleased about it and other posts that got 'moderated' without showing that they had been. Outraged would not cover it.

JohnnyC was not banned

Lie! Proof Below..

@Robbt - A reminder and just for info:

@hairmare commented on 11 Apr • edited
@LibreTime/maintainers These fruitless discussions have lead to me blocking the offending party at the orga level.
I also was copied on the email exchanges which are not reflected in the Git. I could post them here - but it got very childish all around. He had a back-channel into the AT Devs and much of the 2.51 & 2.52 fixes were his and merged in by Albert S.

We did provide a forum as a contribution to the project and BEFORE Haremare suggested making one, which was a reaction to our suggestion. Check the Datestamps. Ours is operational. I wrote it. Where is the Libretime one?
I will happily redirect all traffic to it and save our hosting fees. Please supply the URL. If you do not have one, I don't think we should believe you.

How would cutting the only link to this site, that he paid cash for, benefit him?

The full madness is here:
https://github.com/LibreTime/libretime/issues/156

I think that is team 'Websites International' and our family of stations done here, at least for now. JC may reverse this. If and when we see some maturity, structure, quality control, accountability.. who knows.
I will pass on your good wishes. Both JC and his wife are both being treated for cancer. I am sure we all wish them well.
I think it all comes down to Haremare. The sooner that he realizes that he is just a cog in the wheel and not the whole wheel - the better.

**Comment by Robby262** [comment]
Do not give me the nobody deletes stuff bullshit. You have been caught with your pants down
Which means:
You cannot be trusted as a software partner

**Comment by Robby262** [comment]
Reported

**Comment by Robby262** [comment]
There are meds for this :(
